diamond-api/src/render.rs

```diff
@@ -144,54 +155,280 @@ fn walk(dir: &Path, metric: &str, interval: Interval) -> Result<Vec<RenderPoint>
     Ok(points)
 }
 
+fn walker(w: Walker) -> impl Fn(&str, Interval) -> Result<Vec<RenderPoint>, ResponseError> {
+    move |metric, interval| match &w {
+        Walker::File(dir) => walk(dir.as_path(), metric, interval),
+        Walker::Const(res) => Ok(res.to_vec()),
+    }
+}
+
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 pub struct RenderPoint(Option<f64>, u32);
 
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
-pub struct RenderResponceEntry {
+pub struct RenderResponseEntry {
     target: String,
     datapoints: Vec<RenderPoint>,
 }
 
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
-pub struct RenderResponce {
-    entries: Vec<Option<RenderResponceEntry>>,
+pub struct RenderResponse {
+    entries: Vec<Option<RenderResponseEntry>>,
 }
 
 pub fn render_handler(
-    state: Data<Args>,
-    params: RenderQuery,
+    ctx: Data<Context>,
+    query: RenderQuery,
 ) -> impl Future<Item = HttpResponse, Error = failure::Error> {
-    let dir = &state.path;
-
-    match Interval::new(params.from, params.until).map_err(err_msg) {
+    match Interval::new(query.from, query.until).map_err(err_msg) {
         Ok(interval) => {
-            let response: Result<Vec<RenderResponceEntry>, ResponseError> = params
+            let response: Result<Vec<RenderResponseEntry>, ResponseError> = query
                 .target
                 .into_iter()
-                .map(|x| {
-                    walk(&dir, &x, interval).map(|datapoints| RenderResponceEntry {
-                        datapoints,
-                        target: x,
+                .map(|metric| {
+                    let w = walker(ctx.walker.clone());
+                    w(&metric, interval).map(|points| RenderResponseEntry {
+                        datapoints: points,
+                        target: metric,
                     })
                 })
                 .collect();
 
             match response {
-                Ok(response) => result(Ok(HttpResponse::Ok().json(response))),
+                Ok(response) => result(Ok(format_response(response, query.format))),
                 Err(e) => err(e.into()),
             }
         }
         Err(e) => err(e),
     }
 }
 
+// TODO: Extract code BELOW to response_format_csv module
```

Comment:
What do you guys think, is it worth extracting to a separate module?

Reply:
Maybe we need to add more types and extract them all into the module?

Reply:
I'm not following, what types are you referring to?

Reply:
For example `raw`:
https://github.com/GiantPlantsSociety/diamond/blob/master/diamond-api/src/render.rs#L72-L81